### PR TITLE
[km200] Allow file-defined things via a rootService config parameter

### PR DESCRIPTION
**Pull request by danilobuerger**
Circuit and sensor things read their device service path from the "root" property, which is only ever set by discovery. Things defined in files (.things or YAML) cannot set properties, so they always fail to initialize with "root property missing", making file-based configuration impossible.

When the root property is absent, fall back to a "rootService" config parameter, so the service path (for example /heatingCircuits/hc1) can be given in file configuration.

I am not sure config is the right mechanism here, since root has until now been a property rather than a parameter. I also considered deriving the root from the thing, either mapping thing type and id to the path, or changing the ids so the path is encoded in it. It works for circuits and sensors, but switch-program things would then have to encode their two setpoint names (which are read from the device) into the id, overloading the identifier. I went with a config parameter instead, but I am open to either.

### CODE REVIEW

**Comment by danilobuerger** [comment]
Hi @jlaur could you please review this? (Sorry not sure about the rules here yet, I am new to this repo)

File: bundles/org.openhab.binding.km200/src/main/java/org/openhab/binding/km200/internal/handler/KM200ThingHandler.java [hunk 1]
```diff
@@ -196,6 +196,10 @@ public void initialize() {
             return;
         }
         String service = KM200Utils.translatesNameToPath(thing.getProperties().get("root"));
+        if (service == null) {
+            Object rootService = thing.getConfiguration().get("rootService");
+            service = rootService != null ? rootService.toString() : null;
+        }
```
Comment (lsiepel) [inline bundles/org.openhab.binding.km200/src/main/java/org/openhab/binding/km200/internal/handler/KM200ThingHandler.java:202]: File-defined Things already receive their Thing-type XML properties from openHAB’s [ThingFactory](https://github.com/openhab/openhab-core/blob/c24bbb671536f4aaaff099fca463433b15231c4b/bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/binding/ThingFactory.java#L94-L96). Each affected KM200 type defines a non-null `root`—for example, [heatingCircuit.xml](https://github.com/openhab/openhab-addons/blob/4ff02f51a37aa7e719a13ca231dda9ec81890ff1/bundles/org.openhab.binding.km200/src/main/resources/OH-INF/thing/heatingCircuit.xml#L13-L15)—and `switchProgram` defines it as an empty string. Consequently this condition is false and `[rootService="/heatingCircuits/hc1"]` is ignored. Please give the explicit configuration precedence and fall back to the property. A regression test should construct a Thing with both values so this behavior is covered.

```suggestion:-4+0
        Object rootService = thing.getConfiguration().get("rootService");
        String service = rootService != null ? rootService.toString()
                : KM200Utils.translatesNameToPath(thing.getProperties().get("root"));
```

File: bundles/org.openhab.binding.km200/src/main/java/org/openhab/binding/km200/internal/handler/KM200ThingHandler.java [hunk 2]
```diff
@@ -196,6 +196,10 @@ public void initialize() {
             return;
         }
         String service = KM200Utils.translatesNameToPath(thing.getProperties().get("root"));
+        if (service == null) {
+            Object rootService = thing.getConfiguration().get("rootService");
+            service = rootService != null ? rootService.toString() : null;
```
Comment (lsiepel) [inline bundles/org.openhab.binding.km200/src/main/java/org/openhab/binding/km200/internal/handler/KM200ThingHandler.java:201]: For any `rootService` value that reaches this code, an empty or unknown path fails `containsService`, skips the switch-program branch, then removes all channels and reports the Thing `ONLINE`. Since this path is user-entered, a typo would produce a healthy-looking Thing with no channels. Please validate blank/unknown paths and report `OFFLINE` with `CONFIGURATION_ERROR`, preserving the special switch-program handling.

This edge case should also be covered by a focused test, while non-blocking, this would be appreciated.

File: bundles/org.openhab.binding.km200/src/main/java/org/openhab/binding/km200/internal/handler/KM200ThingHandler.java [hunk 3]
```diff
@@ -196,6 +196,10 @@ public void initialize() {
             return;
         }
         String service = KM200Utils.translatesNameToPath(thing.getProperties().get("root"));
+        if (service == null) {
+            Object rootService = thing.getConfiguration().get("rootService");
```
Comment (lsiepel) [inline bundles/org.openhab.binding.km200/src/main/java/org/openhab/binding/km200/internal/handler/KM200ThingHandler.java:200]: `rootService` is a new public configuration key, but it is absent from the Thing config descriptions, i18n metadata, and README. The existing `.things` example still omits the per-instance path and therefore does not demonstrate the new feature. Please declare and document which Thing types accept this parameter and update the textual/YAML examples accordingly. The [openHAB review checklist](https://github.com/openhab/openhab-addons/wiki/Review-Checklist) requires every configuration key to be documented in the README.

Comment (lsiepel) [reply bundles/org.openhab.binding.km200/src/main/java/org/openhab/binding/km200/internal/handler/KM200ThingHandler.java:200]: Also recommended to extract the magic string to a constant string in `KM200BindingConstants` for the configuration key

**Review by lsiepel — CHANGES_REQUESTED** [review]
Thank you for providing this patch. Me and my codex friend had a look and we come to these review comments.

_This is an initial AI-assisted review._
